openid-4-verifiable-presentations-1_0.md

```diff
@@ -2001,68 +2037,256 @@ Note: The `nonce` and `aud` are set to the `nonce` of the request and the Client
+* Protocol Extension: OpenID for Verifiable Presentations
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#error-response) of this specification
+
+### wallet_unavailable
+
+* Name: `wallet_unavailable`
+* Usage Location: authorization endpoint, token endpoint
+* Protocol Extension: OpenID for Verifiable Presentations
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#error-response) of this specification
+
+## OAuth Authorization Server Metadata Registry
+
+This specification registers the following authorization server metadata parameters
+in the IANA "OAuth Authorization Server Metadata" registry [@IANA.OAuth.Parameters]
+established by [@!RFC8414].
+
+### presentation_definition_uri_supported
+
+* Metadata Name: `presentation_definition_uri_supported`
+* Metadata Description: Boolean value specifying whether the Wallet supports the transfer of presentation_definition by reference
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#as_metadata_parameters) of this specification
+
+### vp_formats_supported
+
+* Metadata Name: `vp_formats_supported`
+* Metadata Description: An object containing a list of name/value pairs, where the name is a string identifying a Credential format supported by the Wallet
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#as_metadata_parameters) of this specification
+
+## OAuth Dynamic Client Registration Metadata Registry
+
+This specification registers the following client metadata parameters
+in the IANA "OAuth Dynamic Client Registration Metadata" registry [@IANA.OAuth.Parameters]
+established by [@!RFC7591].
+
+### vp_formats
+
+* Client Metadata Name: `vp_formats`
+* Client Metadata Description: Boolean value specifying whether the Wallet supports the transfer of presentation_definition by reference
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#client_metadata_parameters) of this specification
+
+## OAuth URI Registry
+
+This specification registers the following URNs
+in the IANA "OAuth URI" registry [@IANA.OAuth.Parameters]
+established by [@!RFC6755].
+
+### urn:ietf:params:oauth:federation
+
+* URN: `urn:ietf:params:oauth:federation`
+* Common Name: OpenID Federation
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#federations) of this specification
+
+### urn:ietf:params:oauth:federation_trust_mark
+
+* URN: `urn:ietf:params:oauth:federation_trust_mark`
+* Common Name: OpenID Federation Trust Mark
+* Change Controller: OpenID Foundation Artifact Binding Working Group - openid-specs-ab@lists.openid.net
+* Reference: (#federations) of this specification
```

Comment:
It seems entirely inappropriate for these to URNs to be in an IETF OAuth namespace.

Reply:
What's an appropriate or inappropriate use of the OAuth URN namespace is likely in the eye of the beholder.  For instance, the[ COSE Key Thumbprint spec uses it](https://www.ietf.org/archive/id/draft-ietf-cose-key-thumbprint-06.html#section-5.6) - paralleling [its use by the JWK Thumbprint spec]( https://www.rfc-editor.org/rfc/rfc9278.html#name-jwk-thumbprint-uri), with there also being [its use in the JWK Thumbprint URI spec](https://www.rfc-editor.org/rfc/rfc9278.html#name-jwk-thumbprint-uri).

But as Elton John said, "Don't Shoot Me I'm Only the Piano Player".

My role in creating this PR was to register the registerable things that the specification already defined.  I didn't try to pass judgement on whether they were the ***right things***.

If you think these URNs already defined at [11.2. ](https://openid.net/specs/openid-4-verifiable-presentations-1_0-21.html#section-11.2)[Support for Federations/Trust Schemes](https://openid.net/specs/openid-4-verifiable-presentations-1_0-21.html#name-support-for-federations-tru) are the ***wrong things***, convince the working group to change them, and I'll gladly update this PR to match.

Reply:
As Billy Joel said in a different song about a piano player, "but there's some place that he'd rather be."

Indeed there is some element to 'eye of the beholder' here but, as co-author of the document that established the IANA "OAuth URI" registry, I behold these to be entirely inappropriate. 

I would furthermore behold that JWK Thumbprint URI spec's use of it is pretty tenuous and the COSE Key Thumbprint spec's use is even more so. But it's way too late to do anything about the former and mostly too late for the latter. 

There seem to be a lot of ***wrong things*** in [11.2. ](https://openid.net/specs/openid-4-verifiable-presentations-1_0-21.html#section-11.2)[Support for Federations/Trust Schemes](https://openid.net/specs/openid-4-verifiable-presentations-1_0-21.html#name-support-for-federations-tru) and I've put in #279 to try and make two of those wrong things less wrong.  

In the meantime, this PR should not exacerbate the situation.

Reply:
I removed the registration of the OAuth URNs since it has become clear that the WG will remove them in some manner.